iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/ReplaceTieLinesByLines.java

```diff
@@ -0,0 +1,131 @@
+            DanglingLine dl2 = tl.getDanglingLine2();
+            String dl1Id = dl1.getId();
+            String dl2Id = dl2.getId();
+            LineAdder adder = network.newLine()
+                    .setId(tl.getId())
+                    .setName(tl.getOptionalName().orElse(null))
+                    .setR(tl.getR())
+                    .setX(tl.getX())
+                    .setB1(tl.getB1())
+                    .setB2(tl.getB2())
+                    .setG1(tl.getG1())
+                    .setG2(tl.getG2())
+                    .setVoltageLevel1(dl1.getTerminal().getVoltageLevel().getId())
+                    .setVoltageLevel2(dl2.getTerminal().getVoltageLevel().getId());
+            if (dl1.getTerminal().getVoltageLevel().getTopologyKind() == TopologyKind.NODE_BREAKER) {
+                adder.setNode1(dl1.getTerminal().getNodeBreakerView().getNode());
+            } else {
+                adder.setConnectableBus1(dl1.getTerminal().getBusBreakerView().getConnectableBus().getId())
+                        .setBus1(Optional.ofNullable(dl1.getTerminal().getBusBreakerView().getBus())
+                                .map(Identifiable::getId)
+                                .orElse(null));
+            }
+            if (dl2.getTerminal().getVoltageLevel().getTopologyKind() == TopologyKind.NODE_BREAKER) {
+                adder.setNode2(dl2.getTerminal().getNodeBreakerView().getNode());
+            } else {
+                adder.setConnectableBus2(dl2.getTerminal().getBusBreakerView().getConnectableBus().getId())
+                        .setBus2(Optional.ofNullable(dl2.getTerminal().getBusBreakerView().getBus())
+                                .map(Identifiable::getId)
+                                .orElse(null));
+            }
+            warningAboutExtensions(dl1, dl2, tl, reporter);
+            TopologyModificationUtils.LoadingLimitsBags limits1 = new TopologyModificationUtils.LoadingLimitsBags(dl1::getActivePowerLimits,
+                    dl1::getApparentPowerLimits, dl1::getCurrentLimits);
+            TopologyModificationUtils.LoadingLimitsBags limits2 = new TopologyModificationUtils.LoadingLimitsBags(dl2::getActivePowerLimits,
+                    dl2::getApparentPowerLimits, dl2::getCurrentLimits);
+            String xNode = tl.getUcteXnodeCode();
+            double p1 = dl1.getTerminal().getP();
+            double q1 = dl1.getTerminal().getQ();
+            double p2 = dl2.getTerminal().getP();
+            double q2 = dl2.getTerminal().getQ();
+            Properties properties = new Properties();
+            tl.getPropertyNames().forEach(pn -> properties.put(pn, tl.getProperty(pn)));
+            mergeProperties(dl1, dl2, properties, reporter);
+            Map<String, String> aliases = new HashMap<>();
+            tl.getAliases().forEach(alias -> aliases.put(alias, tl.getAliasType(alias).orElse("")));
+            mergeIdenticalAliases(dl1, dl2, aliases, reporter);
```

Comment:
There cannot be identical aliases, dl1 and dl2 are in the same network. An exception is thrown when an existing alias is added in the network index.